crates/transaction-pool/src/traits.rs

```diff
@@ -532,6 +532,18 @@ pub struct PooledTransaction {
 }
 
 impl PooledTransaction {
+    /// Create new instance of [Self].
+    pub fn new(transaction: TransactionSignedEcRecovered) -> Self {
+        let gas_cost = match &transaction.transaction {
+            Transaction::Legacy(t) => U256::from(t.gas_price) * U256::from(t.gas_limit),
+            Transaction::Eip2930(t) => U256::from(t.gas_price) * U256::from(t.gas_limit),
+            Transaction::Eip1559(t) => U256::from(t.max_fee_per_gas) * U256::from(t.gas_limit),
+        };
+        let cost = gas_cost + U256::from(transaction.value());
```

Comment:
Will we continue to need this?

Reply:
the cost is hot loaded in pool `update` fn. don't mind leaving this